### ISSUE TITLE
"Remove project folder" doesn't work if executed on a folder that contains only a single subfolder

### ISSUE THREAD

**Issue by DictumMortuum** [atom/tree-view#942]
### Prerequisites
- [x] Can you reproduce the problem in [safe mode](http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode)?
- [x] Are you running the [latest version of Atom](http://flight-manual.atom.io/hacking-atom/sections/debugging/#update-to-the-latest-version)?
- [x] Did you check the [debugging guide](http://flight-manual.atom.io/hacking-atom/sections/debugging/)?
- [x] Did you check the [FAQs on Discuss](https://discuss.atom.io/c/faq)?
- [x] Are you reporting to the [correct repository](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#atom-and-packages)?
- [x] Did you [perform a cursory search](https://github.com/issues?q=is%3Aissue+user%3Aatom+-repo%3Aatom%2Felectron) to see if your bug or enhancement is already reported?

For more information on how to write a good [bug report](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#how-do-i-submit-a-good-bug-report) or [enhancement request](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#how-do-i-submit-a-good-enhancement-suggestion), see the `CONTRIBUTING` guide.
### Description

When I try to remove a project folder that contains only a single subfolder, it doesn't work.
My workaround is to touch a temp file, wait for atom to update the tree view with the new temp file, remove the folder and then delete the temp file.
### Steps to Reproduce
1. mkdir -p /tmp/project/subfolder && cd /tmp/project/subfolder && touch bla
2. add project folder /tmp/project
3. remove project folder

![remove project folder collapsed](https://cloud.githubusercontent.com/assets/1058982/19119103/f667a290-8b1e-11e6-96f1-57d28b8c55bc.gif)

**Expected behavior:**

The project is removed from the tree-view.

**Actual behavior:**

It wasn't.
This might have something to do with the fact that if a folder is empty, it is truncated in the tree view. For instance, it appears like so "project/subfolder" instead of just "project" with the expanding arrow.
### Versions

You can get this information from executing `atom --version` and `apm --version` at the command line. Also, please include the OS and what version of the OS you're running.

```
git:master on apparatus in ~ atom --version
Atom    : 1.10.2
Electron: 0.37.8
Chrome  : 49.0.2623.75
Node    : 5.10.0
git:master on apparatus in ~ apm --version
apm  1.12.5
npm  3.10.5
node 4.4.5
python 2.7.12
git 2.7.4
```

**Comment by preynal** [comment]
I'm having the same issue. To reproduce, set collapse directories in the package settings.

**Comment by Ben3eeE** [comment]
I can reproduce this on Atom `1.12-dev-fcddf83` (Electron 1.4.1 branch).

![remove project folder collapsed](https://cloud.githubusercontent.com/assets/1058982/19119103/f667a290-8b1e-11e6-96f1-57d28b8c55bc.gif)

**Comment by lee-dohm** [comment]
This is a side-effect of #697. The proper fix is to solve #697.

**Comment by lee-dohm** [comment]
Fixed in #1013.
